File: server/internal/dto/base/response/menu.go
package response

// MenuResponse 菜单响应
// 用于返回菜单相关信息，字段名统一使用小写（符合 JSON 规范）
type MenuResponse struct {
	ID           string         `json:"id"`                 // 菜单 ID（字符串格式，使用 Snowflake ID）
	ParentID     string         `json:"parentId"`           // 父菜单 ID（字符串格式，防止 string 精度丢失）
	Path         string         `json:"path"`               // 路由路径
	Name         string         `json:"name"`               // 路由名称
	Component    string         `json:"component"`          // 组件路径
	Title        string         `json:"title"`              // 菜单标题
	Icon         string         `json:"icon"`               // 菜单图标
	Hidden       bool           `json:"hidden"`             // 是否隐藏
	Sort         int            `json:"sort"`               // 排序号
	Status       int            `json:"status"`             // 状态：0 禁用，1 启用
	IsExt        bool           `json:"isExt"`              // 是否外部链接
	Redirect     string         `json:"redirect"`           // 重定向地址
	Permission   string         `json:"permission"`         // 权限标识
	KeepAlive    bool           `json:"keepAlive"`          // 是否缓存
	DefaultMenu  bool           `json:"defaultMenu"`        // 是否默认菜单
	Breadcrumb   bool           `json:"breadcrumb"`         // 是否在面包屑显示
	ActiveMenu   string         `json:"activeMenu"`         // 当前激活的菜单
	Affix        bool           `json:"affix"`              // 是否固定在标签栏
	Type         int            `json:"type"`               // 菜单类型：1 目录，2 菜单，3 按钮，4iframe
	FrameLoading bool           `json:"frameLoading"`       // iframe 加载动画
	Meta         *MenuMetaResp  `json:"meta,omitempty"`     // 菜单元数据
	Children     []MenuResponse `json:"children,omitempty"` // 子菜单
}

// MenuMetaResp 菜单元数据响应
type MenuMetaResp struct {
	Title        string `json:"title"`        // 菜单标题
	Icon         string `json:"icon"`         // 菜单图标
	KeepAlive    bool   `json:"keepAlive"`    // 是否缓存（驼峰命名）
	DefaultMenu  bool   `json:"defaultMenu"`  // 是否默认菜单（驼峰命名）
	Breadcrumb   bool   `json:"breadcrumb"`   // 是否在面包屑显示（驼峰命名）
	ActiveMenu   string `json:"activeMenu"`   // 当前激活的菜单（驼峰命名）
	Affix        bool   `json:"affix"`        // 是否固定在标签栏（驼峰命名）
	FrameLoading bool   `json:"frameLoading"` // iframe 加载动画（驼峰命名）
}

File: web/src/types/menu.ts
// 路由菜单结构，用于前端路由导航
export interface RouterMenuItem {
  id: string | number
  name: string
  path: string
  component: string | (() => Promise<{ default: import('vue').Component }>)
  meta: {
    title: string
    icon: string
    hidden?: boolean
    keepAlive?: boolean
    closeTab?: boolean
    btns?: string[]
    defaultMenu?: boolean
    transitionType?: string
  }
  hidden?: boolean
  children?: RouterMenuItem[]
}

// 菜单类型枚举
export enum MenuType {
  GROUP = 1, // 不可打开页面的菜单（父级菜单，用于分组）
  MENU = 2, // 可以打开页面的菜单（实际可访问的页面）
  BUTTON = 3, // 按钮权限数据（不是菜单，是页面内的操作按钮）
}

// 后端 API 返回的菜单数据结构（所有字段名统一使用小写，符合 JSON 规范）
export interface ApiMenuItem {
  id: string
  parentId: string // 必须使用 string，防止 int64 精度丢失
  path: string
  name: string
  component: string | (() => Promise<{ default: import('vue').Component }>)
  title: string
  icon: string
  hidden: boolean
  sort: number
  status: number
  isExt: boolean
  redirect: string
  permission: string
  keepAlive: boolean
  defaultMenu: boolean
  breadcrumb: boolean
  activeMenu: string
  affix: boolean
  type: MenuType | number // 1: 目录，2: 菜单，3: 按钮
  frameLoading: boolean
  apiIds?: string[]
  meta?: {
    title: string
    icon: string
    keepAlive: boolean
    defaultMenu: boolean
    breadcrumb: boolean
    activeMenu: string
    affix: boolean
    frameLoading: boolean
  }
  children?: ApiMenuItem[]
}

// 按钮权限数据结构
export interface ButtonPermission {
  id: string
  name: string // 按钮名称，如 "UserAdd"
  title: string // 按钮标题，如 "新增用户"
  permission: string // 权限标识，如 "system:user:add"
  icon: string // 按钮图标
  parentId: string // 所属页面菜单的 ID
}

// 菜单删除影响评估
export interface MenuDeleteImpact {
  childMenusCount: number
  btnPermsCount: number
  affectedRolesCount: number
  affectedApisCount: number
  level: 'LOW' | 'MEDIUM' | 'HIGH'
  warning: string
}
